### PR TITLE
Amazon S3 integration and an image infrastructure for it

### PR DESCRIPTION
**Pull request by ghost**
* The docker files have been changed by the removal of unnecessary comments and the insertion of secret variables; 
* The pom.xml has been upgraded to a more productive one; 
* The exception handling of incorrect cloud credentials has been added to the AmazonConfig.java; 
* The logic of the paths of the enviroment variables has been changed to a more convenient one;

### CODE REVIEW

File: src/main/resources/db/changelog/v.0.6/db.changelog-add-discounts-image-data.yaml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            columns:
+              - column:
+                  name: id
+                  value: "9e0c95cd-b007-4d6d-933d-bb23e8532945"
+              - column:
+                  name: discount_id
+                  value: "dd5877e8-8b5a-44dd-9e4f-4675c3ba6adc"
+              - column:
+                  name: image
+                  value: "https://exdadel-discounts-image-upload.s3.eu-central-1.amazonaws.com/discounts-images/nails1.jpg"
```
Comment (Hanna-Kalashnyk) [inline src/main/resources/db/changelog/v.0.6/db.changelog-add-discounts-image-data.yaml:140]: think, put everywhere
 name: by_default
                  value: true

or neither

**Review by Hanna-Kalashnyk — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -31,34 +26,44 @@
...
+        <amazon-aws.version>1.12.21</amazon-aws.version>
+        <google-gson.version>2.8.7</google-gson.version>
         <jwt.version>3.17.0</jwt.version>
-        <json.version>20210307</json.version>
     </properties>
 
+    <parent>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <groupId>org.springframework.boot</groupId>
+        <version>2.3.12.RELEASE</version>
```
Comment (mikevoloshyn) [inline pom.xml:37]: `<version>${spring-boot.version}</version>`

Comment (ghost) [reply pom.xml:37]: When I am trying to put that version through the "${}" in that parent I am catching an error in pom.xml.
image
But when I put the version directly, everything is fine.

Comment (mikevoloshyn) [reply pom.xml:37]: clear. go ahead

**Review by mikevoloshyn — APPROVED** [review]

**Pull request merged**